deps/c-ares/src/ares_getnameinfo.c

```diff
@@ -155,8 +155,12 @@ void ares_getnameinfo(ares_channel channel, const struct sockaddr *sa,
           }
         /* They also want a service */
         if (flags & ARES_NI_LOOKUPSERVICE)
-          service = lookup_service((unsigned short)(port & 0xffff),
+          service = lookup_service(port,
                                    flags, srvbuf, sizeof(srvbuf));
+        else {
```

Comment:
ipv6 resolving never worked (!)

Reply:
Did you take these changes from upstream?

Reply:
NO, I just have coded by hands.

Reply:
I'll take the changes, since we are already in fork territory, but it'd be best if you submit them upstream too: https://github.com/c-ares/c-ares so others can benefit from your improvements.

Reply:
c-ares in master has exactly the same bugs. and terrible code. I can remove tests as well as fixes in c-ares, if you want

Reply:
If it's broken, I'd opt for fixing it upstream too, if you have the inclination, that is. You can leave the fix here and the tests too, to make sure things don't break on the next upstream update.

Reply:
I've added expected failure. With my fixes, they were succeeded (except on Windows where the world is exceptionally broken). That's my final decision.

Reply:
I don't have patience in order to fix Windows bugs. Yep, I can give you patches for Linux. Do you need ?

Reply:
Fair enough.